shielded-pool/src/component.rs

```diff
@@ -491,50 +486,40 @@ pub trait View: StateExt {
                 id,
                 denom: denom.clone(),
             });
-            self.put_domain("shielded_pool/known_assets".into(), known_assets)
+            self.put_domain(state_key::known_assets(), known_assets)
                 .await;
             Ok(())
         }
     }
 
     async fn set_note_source(&self, note_commitment: &note::Commitment, source: NoteSource) {
-        self.put_domain(
-            format!("shielded_pool/note_source/{}", note_commitment).into(),
-            source,
-        )
-        .await
+        self.put_domain(state_key::note_source(note_commitment), source)
+            .await
     }
 
     async fn note_source(&self, note_commitment: &note::Commitment) -> Result<Option<NoteSource>> {
-        self.get_domain(format!("shielded_pool/note_source/{}", note_commitment).into())
+        self.get_domain(state_key::note_source(note_commitment))
             .await
     }
 
     async fn set_compact_block(&self, compact_block: CompactBlock) {
-        self.put_domain(
-            format!("shielded_pool/compact_block/{}", compact_block.height).into(),
-            compact_block,
-        )
-        .await
+        self.put_domain(state_key::set_compact_block(&compact_block), compact_block)
+            .await
     }
 
     async fn compact_block(&self, height: u64) -> Result<Option<CompactBlock>> {
-        self.get_domain(format!("shielded_pool/compact_block/{}", height).into())
-            .await
+        self.get_domain(state_key::compact_block(height)).await
     }
 
     async fn set_nct_anchor(&self, height: u64, nct_anchor: tct::Root) {
         tracing::debug!(?height, ?nct_anchor, "writing anchor");
 
         // Write the NCT anchor both as a value, so we can look it up,
-        self.put_domain(
-            format!("shielded_pool/tct_anchor/{}", height).into(),
-            nct_anchor,
-        )
-        .await;
+        self.put_domain(state_key::nct_anchor_tct(&nct_anchor), nct_anchor)
+            .await;
```

Comment:
This isn't quite the right translation: the key should be derived from the _height_, not the anchor, so that we can query for what the anchor at a given height is.

To fix this, let's replace `nct_anchor_tct(nct_anchor: &Root)` by `anchor_by_height(height: u64)`, to match the original code.